mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,392 @@
+                    get_coords = 0
+        for i_index in range(3):
+            fiducial_coords[i_index] = np.asarray([float(x) for x in fiducial_coords[i_index]])
+
+        ###get coordinates for sources###
+        source_coords = []
+        for i_chan in source_label:
+            if i_chan in all_labels:
+                chan_index = all_labels.index(i_chan)
+                source_coords.append(all_coords[chan_index])
+                
+        ###get coordinates for detectors###
+        detect_coords = []
+        for i_chan in detect_label:
+            if i_chan in all_labels:
+                chan_index = all_labels.index(i_chan)
+                detect_coords.append(all_coords[chan_index])
+                
+
+        # Generate meaningful channel names
+        ###need to rename labels to make other functions happy###
+        ###get our unique labels for sources and detectors###
+        unique_source_labels = []
+        unique_detect_labels = []
+        [unique_source_labels.append(label) for label in source_label if label not in unique_source_labels]
+        [unique_detect_labels.append(label) for label in detect_label if label not in unique_detect_labels]
+
+        ###now let's label each channel in our data###
+        ###data is channels X timepoint where the first source_num rows correspond to###
+        ###the first detector, and each row within that group is a different source###
+        ###should note that current .mtg files contain channels for multiple data files###
+        ###going to move to have a single .mtg file per participant, condition, and montage###
+        ###combine coordinates and label our channels###
+        ###will label them based on ac, dc, and ph data###
+        boxy_coords = []
+        boxy_labels = []
+        data_types = ['AC','DC','Ph']
+        total_chans = detect_num*source_num
+        for i_type in data_types:
+            for i_coord in range(len(source_coords[0:total_chans])):
+                boxy_coords.append(np.mean(
+                    np.vstack((source_coords[i_coord], detect_coords[i_coord])),
+                    axis=0).tolist() + source_coords[i_coord] + 
+                    detect_coords[i_coord] + [chan_wavelength[i_coord]] + [0] + [0])
+                boxy_labels.append('S' + 
+                                       str(unique_source_labels.index(source_label[i_coord])+1)
+                                       + '_D' + 
+                                       str(unique_detect_labels.index(detect_label[i_coord])+1) 
+                                       + ' ' + chan_wavelength[i_coord] + ' ' + i_type)
+        
+        # add extra column for triggers
+        boxy_labels.append('Markers')
+
+        ###montage only wants channel coords, so need to grab those, convert to###
+        ###array, then make a dict with labels###
+        for i_chan in range(len(boxy_coords)):
+            boxy_coords[i_chan] = np.asarray(boxy_coords[i_chan],dtype=np.float64) 
```

Comment:
You should not need to nest this in a loop
```suggestion
        boxy_coords = np.array(boxy_coords, float)
        assert boxy_coords.shape == (len(boxy_coords), 12)
```
Or better, above preallocate `boxy_coords = np.zeros((n_expected, 12))` and then populate `boxy_coords` directly

Reply:
went with the first to not go down a preallocation rabbit hole (need to do labels too)